### ISSUE TITLE
ASTF/STL multi-profile is not stable 

### ISSUE THREAD

**Issue by hhaim** [cisco-system-traffic-generator/trex-core#360]
The tests are going to be disabled for now 

```
(NOT test.result:PASS) AND (test.name:test_random_dynamic_add_remove)
(NOT test.result:PASS) AND (test.name:test_stress_start_stop_dynamic_profile)
(NOT test.result:PASS) AND (test.name:test_double_start_stop)
```

```
stateless_tests.stl_client_test.STLClient_Test.test_random_dynamic_add_remove
astf_tests.astf_resilience_test.ASTFResilience_Test.test_double_start_stop
```

```
id_input)
  File "/tmp/trex-scripts/automation/trex_control_plane/interactive/trex/astf/stats/traffic.py", line 180, in _get_stats_values
    raise TRexError(rc.err())
TRexError: Invalid profile : febanjrcak
```

Kibana results 
![image](https://user-images.githubusercontent.com/13034429/67632781-f59ab480-f8b0-11e9-9b50-db6a4dcdfd15.png)

@jsmoon could you have a look

**Comment by hhaim** [comment]
Need to revert this commit:

https://github.com/cisco-system-traffic-generator/trex-c
core/commit/5f3c53b2057f14ef15ffdfc1e75eba0e1751c551

It  seems that ```test_double_start_stop``` stability issue is rather new with one of the last commits

**Comment by hhaim** [comment]
Example of the failures 
```
Failure:
Traceback (most recent call last):
File "/usr/lib64/python3.6/unittest/case.py", line 60, in testPartExecutor
yield
File "/usr/lib64/python3.6/unittest/case.py", line 622, in run
testMethod()
File "/tmp/trex-scripts/automation/regression/stateless_tests/stl_client_test.py", line 1072, in test_random_dynamic_add_remove
test.run_test()
File "/tmp/trex-scripts/automation/regression/stateless_tests/stl_client_test.py", line 143, in run_test
assert dropped_packets < 2
AssertionError


Stdout:
1 new profile 0.profile_0 44 1
3 new profile 0.profile_1 34 2
5 new profile 0.profile_2 48 3
7 new profile 0.profile_3 12 4
9 new profile 0.profile_4 26 5
11 new profile 0.profile_5 9 6
13 new profile 0.profile_6 15 7
15 new profile 0.profile_7 41 8
17 new profile 0.profile_8 33 9
19 del profile 3 8
19 new profile 0.profile_9 5 9
20 del profile 5 8
21 new profile 0.profile_10 29 9
23 new profile 0.profile_11 38 10
24 del profile 9 9
25 new profile 0.profile_12 13 10
27 del profile 6 9
27 new profile 0.profile_13 26 10
29 new profile 0.profile_14 28 11
31 new profile 0.profile_15 34 12
33 del profile 4 11
33 new profile 0.profile_16 15 12
35 del profile 1 11
35 new profile 0.profile_17 29 12
37 new profile 0.profile_18 28 13
38 del profile 12 12
39 new profile 0.profile_19 14 13
41 new profile 0.profile_20 9 14
42 del profile 0 13
43 new profile 0.profile_21 2 14
45 new profile 0.profile_22 12 15
46 del profile 21 14
47 del profile 16 13
47 new profile 0.profile_23 25 14
48 del profile 8 13
48 del profile 10 12
49 new profile 0.profile_24 3 13
50 del profile 2 12
50 del profile 20 11
51 new profile 0.profile_25 37 12
52 del profile 13 11
52 del profile 24 10
53 del profile 19 9
53 del profile 7 8
53 new profile 0.profile_26 24 9
55 del profile 14 8
55 new profile 0.profile_27 46 9
57 del profile 22 8
57 new profile 0.profile_28 33 9
58 del profile 11 8
59 new profile 0.profile_29 36 9
61 new profile 0.profile_30 21 10
62 del profile 17 9
63 del profile 15 8
63 new profile 0.profile_31 37 9
64 del profile 18 8
65 new profile 0.profile_32 1 9
67 del profile 32 8
67 new profile 0.profile_33 27 9
69 new profile 0.profile_34 28 10
71 del profile 23 9
71 new profile 0.profile_35 26 10
73 new profile 0.profile_36 11 11
75 new profile 0.profile_37 8 12
76 del profile 26 11
77 new profile 0.profile_38 29 12
79 new profile 0.profile_39 20 13
81 del profile 30 12
81 new profile 0.profile_40 17 13
83 del profile 37 12
83 new profile 0.profile_41 27 13
84 del profile 36 12
85 new profile 0.profile_42 20 13
86 del profile 25 12
87 new profile 0.profile_43 19 13
88 del profile 28 12
89 new profile 0.profile_44 18 13
91 new profile 0.profile_45 3 14
93 del profile 33 13
93 del profile 29 12
93 new profile 0.profile_46 8 13
94 del profile 45 12
95 new profile 0.profile_47 17 13
96 del profile 34 12
96 del profile 35 11
97 del profile 40 10
97 new profile 0.profile_48 21 11
98 del profile 31 10
98 del profile 39 9
98 del profile 27 8
99 new profile 0.profile_49 13 9
101 del profile 46 8
101 new profile 0.profile_50 1 9
102 del profile 50 8
103 new profile 0.profile_51 46 9
104 del profile 42 8
105 del profile 38 7
105 del profile 43 6
105 new profile 0.profile_52 16 7
106 del profile 44 6
107 new profile 0.profile_53 21 7
109 del profile 41 6
109 new profile 0.profile_54 46 7
111 del profile 47 6
111 new profile 0.profile_55 38 7
112 del profile 49 6
113 new profile 0.profile_56 21 7
115 new profile 0.profile_57 5 8
117 del profile 48 7
117 new profile 0.profile_58 45 8
119 new profile 0.profile_59 3 9
120 del profile 57 8
120 del profile 52 7
123 del profile 59 6
128 del profile 53 5
134 del profile 56 4
148 del profile 51 3
149 del profile 55 2
155 del profile 54 1
162 del profile 58 0
stop
[]
dropped_packets 176

>>>>>>>>>>>>>>> Daemon log <<<<<<<<<<<<<<<
10-27 09:20:02 trex_server INFO Processing get_trex_log() command.
10-27 09:22:54 trex_server INFO Processing get_trex_daemon_log() command.

```

![image](https://user-images.githubusercontent.com/13034429/67634374-a27e2d00-f8c3-11e9-9e23-ecc5b4b12b00.png)

![image](https://user-images.githubusercontent.com/13034429/67634380-baee4780-f8c3-11e9-83fb-d22e4f1a9c2b.png)

**Comment by hhaim** [comment]
```double_start_stop``` issue started a few days ago. I don't see a commit relevant for that so I'm assuming something else trigger this 

![image](https://user-images.githubusercontent.com/13034429/67634631-356c9680-f8c7-11e9-9a76-71f5eb5afa4a.png)

**Comment by jsmoon** [comment]
@hhaim Regarding the ASTF issue `TRexError: Invalid profile : febanjrcak`, I found a missing point at `TrexAstf::update_astf_state()`.
During a dynamic profile is in STATE_DELETE state, ASTF global state will be STATE_IDLE. This mismatch between global and profile states might cause ASTFClient's misbehavior. I think this can cause this issue.

**Comment by hhaim** [comment]
Could you commit it and re-enable the test?
The double_start_stop is something new from latest commits (maybe Reset speedup)

**Comment by jsmoon** [comment]
I'm in my home. So I can do it tomorrow.

**Comment by hhaim** [comment]
Revert improve the issue but does not solve it
https://github.com/cisco-system-traffic-generator/trex-core/commit/4a01befadb8c31bd6989877e6f0107d26a5fb87e

**Comment by hhaim** [comment]
solved #362
